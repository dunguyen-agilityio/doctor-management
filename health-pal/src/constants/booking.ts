export const BookingQueryKey = {
  hospital: 'populate[doctor][populate][clinic][fields]',
  patient: 'populate[patient][populate][users_permissions_user][populate][avatar][fields]',
  doctor: 'populate[doctor][populate][users_permissions_user][populate][avatar][fields]',
  specialty: 'populate[doctor][populate][specialty][fields]',
}

export const BookingMessages = {
  BOOKING_SUCCESS: 'Booking successfully created',
  BOOKING_ERROR: 'Failed to create booking',
  BOOKING_CANCEL_SUCCESS: 'Booking successfully cancelled',
  BOOKING_CANCEL_ERROR: 'Failed to cancel booking',
  BOOKING_UPDATE_SUCCESS: 'Booking successfully updated',
  BOOKING_UPDATE_ERROR: 'Failed to update booking',
}
