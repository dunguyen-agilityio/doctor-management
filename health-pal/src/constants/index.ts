export * from './device'
export * from './environment'
export * from './message'
export * from './regex'
export * from './route'
export * from './specialty'
