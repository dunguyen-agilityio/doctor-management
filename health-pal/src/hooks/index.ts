export * from './use-app-loading'
