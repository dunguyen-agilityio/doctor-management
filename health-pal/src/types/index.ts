export * from './auth'
export * from './booking'
export * from './doctor'
export * from './favorite'
export * from './image'
export * from './modal'
export * from './route'
export * from './specialty'
export * from './strapi'
