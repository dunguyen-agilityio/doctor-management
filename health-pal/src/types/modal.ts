export type ModalRef<T = null> = { open: (params?: T) => void; close: () => void }
