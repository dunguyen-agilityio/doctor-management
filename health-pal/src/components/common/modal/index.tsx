import { useEffect } from 'react'

import { Button, Dialog, DialogContentProps, DialogProps, Unspaced } from 'tamagui'

import { WINDOW_SIZE } from '@app/constants'

import { useModal } from '@app/hooks/use-modal'

import { X } from '@icons'

import { ModalRef } from '@app/types/modal'

export interface ModalProps extends DialogContentProps, Omit<DialogProps, 'children'> {
  closeButtonShown?: boolean
  ref?: React.Ref<ModalRef>
}

const Modal = ({
  children,
  closeButtonShown,
  ref,
  open: defaultOpen,
  ...props
}: React.PropsWithChildren<ModalProps>) => {
  const [open, setOpen] = useModal(ref)

  useEffect(() => {
    if (defaultOpen) {
      setOpen(true)
    }
  }, [defaultOpen, setOpen])

  return (
    <Dialog modal open={open} onOpenChange={setOpen}>
      <Dialog.Portal>
        <Dialog.Overlay
          testID="overlay"
          backgroundColor="$shadow3"
          animateOnly={['transform', 'opacity']}
          animation={[
            'quicker',
            {
              opacity: {
                overshootClamping: true,
              },
            },
          ]}
          enterStyle={{ opacity: 0, scale: 0.95 }}
          exitStyle={{ opacity: 0 }}
        />

        <Dialog.Content
          role="dialog"
          testID="content"
          padding={0}
          width={WINDOW_SIZE.width - 48}
          minHeight={WINDOW_SIZE.height / 2}
          borderRadius="$2"
          overflow="hidden"
          shadowColor="#fff"
          shadowOffset={{ height: 0, width: 0 }}
          key="content"
          animateOnly={['transform', 'opacity']}
          animation={[
            'quicker',
            {
              opacity: {
                overshootClamping: true,
              },
            },
          ]}
          enterStyle={{ x: 0, y: -20, opacity: 0 }}
          exitStyle={{ x: 0, y: 10, opacity: 0, scale: 0.95 }}
          {...props}>
          {children}
          {closeButtonShown && (
            <Unspaced>
              <Dialog.Close asChild>
                <Button
                  position="absolute"
                  top="$md"
                  right="$md"
                  size="$5"
                  circular
                  icon={X}
                  aria-label="Close the dialog"
                  accessibilityHint="Closes the cancellation dialog without making changes"
                  role="button"
                />
              </Dialog.Close>
            </Unspaced>
          )}
        </Dialog.Content>
      </Dialog.Portal>
    </Dialog>
  )
}

export default Modal
