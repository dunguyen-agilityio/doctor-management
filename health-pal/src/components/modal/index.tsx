import { useImperativeHandle, useState } from 'react'

import { X } from '@tamagui/lucide-icons'
import { Button, Dialog, DialogContentProps, DialogProps, Unspaced } from 'tamagui'

import { WINDOW_SIZE } from '@app/constants'

import { ModalRef } from '@app/types/modal'

interface ModalProps extends DialogProps {
  closeButtonShown?: boolean
  contentProps?: DialogContentProps
  ref?: React.Ref<ModalRef>
}

const Modal = ({
  children,
  closeButtonShown,
  contentProps,
  ref,
  ...props
}: React.PropsWithChildren<ModalProps>) => {
  const [open, setOpen] = useState(false)

  useImperativeHandle(ref, () => ({
    open: () => setOpen(true),
    close: () => setOpen(false),
  }))

  return (
    <Dialog modal open={open} onOpenChange={setOpen} {...props}>
      <Dialog.Portal>
        <Dialog.Overlay
          key="overlay"
          backgroundColor="$shadow3"
          animateOnly={['transform', 'opacity']}
          animation={[
            'quicker',
            {
              opacity: {
                overshootClamping: true,
              },
            },
          ]}
          enterStyle={{ opacity: 0, scale: 0.95 }}
          exitStyle={{ opacity: 0 }}
        />

        <Dialog.Content
          padding={0}
          width={WINDOW_SIZE.width - 48}
          minHeight={WINDOW_SIZE.height / 2}
          borderRadius="$2"
          overflow="hidden"
          shadowColor="#fff"
          shadowOffset={{ height: 0, width: 0 }}
          key="content"
          animateOnly={['transform', 'opacity']}
          animation={[
            'quicker',
            {
              opacity: {
                overshootClamping: true,
              },
            },
          ]}
          enterStyle={{ x: 0, y: -20, opacity: 0 }}
          exitStyle={{ x: 0, y: 10, opacity: 0, scale: 0.95 }}
          {...contentProps}>
          {children}
          {closeButtonShown && (
            <Unspaced>
              <Dialog.Close asChild>
                <Button position="absolute" top="$md" right="$md" size="$5" circular icon={X} />
              </Dialog.Close>
            </Unspaced>
          )}
        </Dialog.Content>
      </Dialog.Portal>
    </Dialog>
  )
}

export default Modal
