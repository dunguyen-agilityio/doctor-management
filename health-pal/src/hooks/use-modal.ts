import { useImperativeHandle, useState } from 'react'

import { ModalRef } from '@/types/modal'

export const useModal = (ref?: React.Ref<ModalRef | null>) => {
  const [open, setOpen] = useState(false)

  useImperativeHandle(ref, () => ({
    open: () => setOpen(true),
    close: () => setOpen(false),
  }))

  return [open, setOpen] as const
}
