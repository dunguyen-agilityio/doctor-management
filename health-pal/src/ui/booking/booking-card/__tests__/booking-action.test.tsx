import { fireEvent, render, screen } from '@utils-test'

import dayjs from 'dayjs'
import { router } from 'expo-router'

import { ROUTES } from '@/constants'

import * as MockModal from '@/hooks/use-modal'

import { BOOKING_TABS } from '@/types'
import { ModalRef } from '@/types/modal'

import BookingAction from '../booking-action'

jest.mock('expo-router')

describe('BookingAction', () => {
  let setOpenMock: jest.Mock
  const mockRef: React.RefObject<ModalRef> = {
    current: {
      open: jest.fn(),
      close: jest.fn(),
    },
  }
  const defaultProps = {
    doctorName: 'Dr. John Doe',
    doctorId: 1,
    date: dayjs('2025-07-01'),
    doctorDocId: 'doc-ref1',
    documentId: 'booking1',
    type: BOOKING_TABS.UPCOMING,
  }

  beforeEach(() => {
    jest.clearAllMocks()
    setOpenMock = jest.fn()
    jest.spyOn(MockModal, 'useModal').mockImplementation((ref) => {
      if (ref && ref !== null && 'current' in ref) {
        ref.current = mockRef.current
      }

      return [true, setOpenMock]
    })
  })

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('renders buttons for UPCOMING type', () => {
    render(<BookingAction {...defaultProps} />)

    expect(
      screen.getByLabelText('Cancel booking with Dr. John Doe on 2025-07-01 - 00:00 AM'),
    ).toBeTruthy()
    expect(screen.getByLabelText('Reschedule booking with Dr. John Doe')).toBeTruthy()
    expect(screen.getByTestId('cancel-booking-modal')).toBeTruthy()
  })

  it('renders buttons for COMPLETED type', () => {
    render(<BookingAction {...defaultProps} type={BOOKING_TABS.COMPLETED} />)

    expect(screen.getByLabelText('Re-book appointment with Dr. John Doe')).toBeTruthy()
    expect(screen.getByLabelText('Add a review for Dr. John Doe')).toBeTruthy()
    expect(screen.queryByTestId('cancel-booking-modal')).not.toBeTruthy()
  })

  it('renders nothing for CANCELED type', () => {
    render(<BookingAction {...defaultProps} type={BOOKING_TABS.CANCELED} />)

    expect(screen.queryByTestId('mock-button')).not.toBeTruthy()
    expect(screen.queryByTestId('cancel-booking-modal')).not.toBeTruthy()
  })

  it('navigates to booking screen on Re-Book button press', () => {
    render(<BookingAction {...defaultProps} type={BOOKING_TABS.COMPLETED} />)

    const reBookButton = screen.getByLabelText('Re-book appointment with Dr. John Doe')
    fireEvent.press(reBookButton)

    expect(router.navigate).toHaveBeenCalledWith({
      pathname: ROUTES.BOOKING,
      params: { doctorId: 1 },
    })
  })

  it('navigates to doctor screen on Add Review button press', () => {
    render(<BookingAction {...defaultProps} type={BOOKING_TABS.COMPLETED} />)

    const reviewButton = screen.getByLabelText('Add a review for Dr. John Doe')
    fireEvent.press(reviewButton)

    expect(router.navigate).toHaveBeenCalledWith({
      pathname: ROUTES.DOCTOR,
      params: { id: 1 },
    })
  })

  it('opens cancel modal on Cancel button press', () => {
    render(<BookingAction {...defaultProps} />)

    const cancelButton = screen.getByLabelText(
      'Cancel booking with Dr. John Doe on 2025-07-01 - 00:00 AM',
    )
    fireEvent.press(cancelButton)

    expect(mockRef.current?.open).toHaveBeenCalled()
  })

  it('navigates to booking screen with params on Reschedule button press', () => {
    render(<BookingAction {...defaultProps} />)

    const rescheduleButton = screen.getByLabelText('Reschedule booking with Dr. John Doe')
    fireEvent.press(rescheduleButton)

    expect(router.navigate).toHaveBeenCalledWith({
      pathname: ROUTES.BOOKING,
      params: {
        doctorId: 1,
        doctorDocId: 'doc-ref1',
        bookingId: 'booking1',
        date: dayjs('2025-07-01').toISOString(),
      },
    })
  })

  it('has correct accessibility attributes', () => {
    render(<BookingAction {...defaultProps} />)

    const cancelButton = screen.getByLabelText(
      'Cancel booking with Dr. John Doe on 2025-07-01 - 00:00 AM',
    )
    const rescheduleButton = screen.getByLabelText('Reschedule booking with Dr. John Doe')

    expect(cancelButton).toHaveProp(
      'aria-label',
      'Cancel booking with Dr. John Doe on 2025-07-01 - 00:00 AM',
    )
    expect(cancelButton).toHaveProp(
      'accessibilityHint',
      'Opens a confirmation dialog to cancel this appointment',
    )
    expect(rescheduleButton).toHaveProp('aria-label', 'Reschedule booking with Dr. John Doe')
    expect(rescheduleButton).toHaveProp(
      'accessibilityHint',
      'Navigates to the booking screen to change the date or time of this appointment',
    )
  })
})
