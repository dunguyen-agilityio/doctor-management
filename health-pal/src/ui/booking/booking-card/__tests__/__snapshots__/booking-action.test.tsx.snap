// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingAction matches snapshot for COMPLETED type 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    style={[]}
  />,
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 10,
      }
    }
    testID="booking-action"
  >
    <View
      accessibilityHint="Navigates to the booking screen to schedule a new appointment with this doctor"
      aria-label="Re-book appointment with Dr. John Doe"
      focusVisibleStyle={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e5e7eb",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "borderTopWidth": 1,
          "color": "#1c2a3a",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "fontWeight": "500",
          "height": 37,
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <Text
        lineBreakMode="clip"
        numberOfLines={1}
        style={
          {
            "color": "#1c2a3a",
            "flexGrow": 0,
            "flexShrink": 1,
            "fontFamily": "Inter_500Medium",
            "fontSize": 14,
            "lineHeight": 21,
            "userSelect": "none",
          }
        }
        suppressHighlighting={true}
      >
        Re-Book
      </Text>
    </View>
    <View
      accessibilityHint="Navigates to the review screen to submit feedback for this doctor"
      aria-label="Add a review for Dr. John Doe"
      focusVisibleStyle={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1c2a3a",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "borderTopWidth": 1,
          "color": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "fontWeight": "500",
          "height": 37,
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <Text
        lineBreakMode="clip"
        numberOfLines={1}
        style={
          {
            "color": "#ffffff",
            "flexGrow": 0,
            "flexShrink": 1,
            "fontFamily": "Inter_500Medium",
            "fontSize": 14,
            "lineHeight": 21,
            "userSelect": "none",
          }
        }
        suppressHighlighting={true}
      >
        Add Review
      </Text>
    </View>
  </View>,
]
`;

exports[`BookingAction matches snapshot for UPCOMING type 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    style={[]}
  />,
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 10,
      }
    }
    testID="booking-action"
  >
    <View
      accessibilityHint="Opens a confirmation dialog to cancel this appointment"
      aria-label="Cancel booking with Dr. John Doe on 2025-07-01 - 00:00 AM"
      focusVisibleStyle={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e5e7eb",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "borderTopWidth": 1,
          "color": "#1c2a3a",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "fontWeight": "500",
          "height": 37,
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <Text
        lineBreakMode="clip"
        numberOfLines={1}
        style={
          {
            "color": "#1c2a3a",
            "flexGrow": 0,
            "flexShrink": 1,
            "fontFamily": "Inter_500Medium",
            "fontSize": 14,
            "lineHeight": 21,
            "userSelect": "none",
          }
        }
        suppressHighlighting={true}
      >
        Cancel
      </Text>
    </View>
    <View
      accessibilityHint="Navigates to the booking screen to change the date or time of this appointment"
      aria-label="Reschedule booking with Dr. John Doe"
      focusVisibleStyle={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1c2a3a",
          "borderBottomColor": "transparent",
          "borderBottomLeftRadius": 60,
          "borderBottomRightRadius": 60,
          "borderBottomWidth": 1,
          "borderLeftColor": "transparent",
          "borderLeftWidth": 1,
          "borderRightColor": "transparent",
          "borderRightWidth": 1,
          "borderStyle": "solid",
          "borderTopColor": "transparent",
          "borderTopLeftRadius": 60,
          "borderTopRightRadius": 60,
          "borderTopWidth": 1,
          "color": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "fontWeight": "500",
          "height": 37,
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <Text
        lineBreakMode="clip"
        numberOfLines={1}
        style={
          {
            "color": "#ffffff",
            "flexGrow": 0,
            "flexShrink": 1,
            "fontFamily": "Inter_500Medium",
            "fontSize": 14,
            "lineHeight": 21,
            "userSelect": "none",
          }
        }
        suppressHighlighting={true}
      >
        Reschedule
      </Text>
    </View>
  </View>,
  <View
    pointerEvents="auto"
    style={
      {
        "alignItems": "center",
        "bottom": 0,
        "flexDirection": "column",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="auto"
      style={
        {
          "backgroundColor": "rgba(0,0,0,0.16)",
          "bottom": 0,
          "flexDirection": "column",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": [
            {
              "scale": 0.95,
            },
          ],
          "zIndex": 99999,
        }
      }
      testID="overlay"
    />
    <View
      accessibilityHint="Dialog to confirm or keep this appointment"
      aria-describedby="description-:r6t:"
      aria-label="Confirm cancellation of booking with Dr. John Doe on 2025-07-01 - 00:00 AM"
      aria-labelledby="title-:r6t:"
      collapsable={false}
      id="content-:r6t:"
      role="dialog"
      style={
        {
          "backgroundColor": "hsla(0, 0%, 100%, 1)",
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "flexDirection": "column",
          "minHeight": 667,
          "opacity": 0,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "position": "relative",
          "shadowColor": "rgb(255,255,255)",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.0392156862745098,
          "shadowRadius": 10,
          "transform": [
            {
              "translateY": -20,
            },
            {
              "translateX": 0,
            },
            {
              "scale": 1,
            },
          ],
          "width": 702,
          "zIndex": 100000,
        }
      }
      testID="cancel-booking-modal"
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "column",
            "gap": 32,
            "paddingBottom": 32,
            "paddingLeft": 42,
            "paddingRight": 42,
            "paddingTop": 32,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#e0f7f9",
              "borderBottomLeftRadius": 100,
              "borderBottomRightRadius": 100,
              "borderTopLeftRadius": 100,
              "borderTopRightRadius": 100,
              "flexDirection": "column",
              "height": 100,
              "justifyContent": "center",
              "width": 100,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="24"
            bbWidth="24"
            color="$teal"
            fill="none"
            focusable={false}
            height="24"
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width="24"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="2"
            >
              <RNSVGPath
                d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
              <RNSVGPath
                d="M22 4 12 14.01 9 11.01"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          style={
            {
              "color": "#1c2a3a",
              "fontFamily": "Inter_700Bold",
              "fontSize": 20,
              "lineHeight": 30,
            }
          }
          suppressHighlighting={true}
        >
          Confirm Cancellation
        </Text>
        <Text
          style={
            {
              "color": "#1c2a3a",
              "fontFamily": "Inter_400Regular",
              "fontSize": 14,
              "lineHeight": 21,
              "textAlign": "center",
            }
          }
          suppressHighlighting={true}
        >
          Are you sure you want to cancel your appointment with
           
          <Text
            style={
              {
                "color": "#1c2a3a",
                "fontFamily": "Inter_700Bold",
                "fontSize": 14,
                "lineHeight": 21,
              }
            }
            suppressHighlighting={true}
          >
            Dr. John Doe
          </Text>
           
          on
           
          <Text
            style={
              {
                "color": "#1c2a3a",
                "fontFamily": "Inter_700Bold",
                "fontSize": 14,
                "lineHeight": 21,
              }
            }
            suppressHighlighting={true}
          >
            2025-07-01
          </Text>
          ?
        </Text>
        <View
          style={
            {
              "backgroundColor": "white",
              "flexDirection": "column",
              "gap": 10,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityHint="Cancels the appointment and navigates to cancelled bookings"
            aria-label="Confirm cancellation of appointment with Dr. John Doe"
            focusVisibleStyle={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#1c2a3a",
                "borderBottomColor": "transparent",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderBottomWidth": 1,
                "borderLeftColor": "transparent",
                "borderLeftWidth": 1,
                "borderRightColor": "transparent",
                "borderRightWidth": 1,
                "borderStyle": "solid",
                "borderTopColor": "transparent",
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "borderTopWidth": 1,
                "color": "#ffffff",
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "fontWeight": "500",
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 12,
              }
            }
          >
            <Text
              lineBreakMode="clip"
              numberOfLines={1}
              style={
                {
                  "color": "#ffffff",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "fontFamily": "Inter_500Medium",
                  "fontSize": 14,
                  "lineHeight": 21,
                  "userSelect": "none",
                }
              }
              suppressHighlighting={true}
            >
              Yes, Cancel
            </Text>
          </View>
          <View
            accessibilityHint="Closes the dialog and keeps the appointment"
            aria-label="Keep appointment with Dr. John Doe"
            focusVisibleStyle={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomColor": "transparent",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderBottomWidth": 0,
                "borderLeftColor": "transparent",
                "borderLeftWidth": 0,
                "borderRightColor": "transparent",
                "borderRightWidth": 0,
                "borderStyle": "solid",
                "borderTopColor": "transparent",
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "borderTopWidth": 0,
                "color": "#4d9b91",
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "fontWeight": "500",
                "height": 48,
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 12,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 12,
              }
            }
          >
            <Text
              lineBreakMode="clip"
              numberOfLines={1}
              style={
                {
                  "color": "#4d9b91",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "fontFamily": "Inter_500Medium",
                  "fontSize": 14,
                  "lineHeight": 21,
                  "userSelect": "none",
                }
              }
              suppressHighlighting={true}
            >
              No, Keep Appointment
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
